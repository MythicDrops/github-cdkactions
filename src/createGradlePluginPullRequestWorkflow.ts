import { Job, Stack, Workflow } from "cdkactions";
import dedent from "ts-dedent";

export const createGradlePluginPullRequestWorkflow = (
  stack: Stack
): Workflow => {
  const workflow = new Workflow(stack, "pull-request", {
    name: "Pull Request",
    on: {
      pullRequest: { branches: ["main"] },
      push: { branchesIgnore: ["main"] },
    },
  });
  new Job(workflow, "pull-request", {
    name: "CI",
    runsOn: "ubuntu-latest",
    strategy: {
      matrix: {
        gradle: ["6.7.1", "6.8.2"],
        java: ["1.8", "11", "15"],
      },
    },
    steps: [
      {
        name: "Check-out Code",
        uses: "actions/checkout@v2",
        with: { "fetch-depth": 0 },
      },
      {
        name: "Set up JDK",
        uses: "actions/setup-java@v1",
        with: { "java-version": `\${{ matrix.java }}` },
      },
      {
        name: "Make Gradle Wrapper Executable",
        run: dedent`chmod +x ./gradlew`,
      },
      {
        name: "Set Gradle Version",
        run: dedent`./gradlew wrapper --gradle-version \${{ matrix.gradle }}`,
      },
      {
        name: "Test Plugin Directly with Gradle",
        run: dedent`../gradlew check`,
        workingDirectory: "pixeloutlaw-gradle-plugin",
      },
      {
        name: "Test Plugin Application with Gradle",
        run: dedent`./gradlew check`,
      },
    ],
  });
  return workflow;
};
