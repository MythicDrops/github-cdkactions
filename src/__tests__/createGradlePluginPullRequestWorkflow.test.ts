import { App, Stack } from "cdkactions";
import { createGradlePluginPullRequestWorkflow } from "../createGradlePluginPullRequestWorkflow";

describe("createGradlePluginPullRequestWorkflow", () => {
  let app: App;
  let stack: Stack;

  beforeEach(() => {
    app = new App();
    stack = new Stack(app, "cdk");
  });

  it("generates a GitHub Actions workflow", () => {
    const pluginName = "pixeloutlaw-gradle-plugin";
    expect(
      createGradlePluginPullRequestWorkflow(stack, {
        pluginName,
      }).toGHAction()
    ).toMatchSnapshot();
  });
});
