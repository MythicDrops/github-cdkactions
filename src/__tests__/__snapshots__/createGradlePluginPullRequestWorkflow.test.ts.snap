// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createGradlePluginPullRequestWorkflow generates a GitHub Actions workflow 1`] = `
Object {
  "jobs": Object {
    "pull-request": Object {
      "name": "CI",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "name": "Check-out Code",
          "uses": "actions/checkout@v2",
          "with": Object {
            "fetch-depth": 0,
          },
        },
        Object {
          "name": "Set up JDK",
          "uses": "actions/setup-java@v2",
          "with": Object {
            "distribution": "adopt",
            "java-version": "\${{ matrix.java }}",
          },
        },
        Object {
          "name": "Make Gradle Wrapper Executable",
          "run": "chmod +x ./gradlew",
        },
        Object {
          "name": "Set Gradle Version",
          "run": "./gradlew wrapper --gradle-version \${{ matrix.gradle }}",
        },
        Object {
          "name": "Test Plugin Directly with Gradle",
          "run": "../gradlew check",
          "working-directory": "pixeloutlaw-gradle-plugin",
        },
        Object {
          "name": "Test Plugin Application with Gradle",
          "run": "./gradlew check",
        },
      ],
      "strategy": Object {
        "matrix": Object {
          "gradle": Array [
            "7.0.2",
          ],
          "java": Array [
            "1.8",
            "11",
            "15",
          ],
        },
      },
    },
  },
  "name": "Pull Request",
  "on": Object {
    "pull_request": Object {
      "branches": Array [
        "main",
      ],
    },
    "push": Object {
      "branches-ignore": Array [
        "main",
      ],
    },
  },
}
`;
